<font size = "6">**INTRODUCTION**</font>

My friend Pino Bacinelo is the President and Managing Broker of Pacific M&A and Business Brokers Ltd.<br>
A business broker is the professional advisor/intermediary who helps business owners sell their business. <br>
If anyone reading this notebook is thinking of selling his business in British Columbia, Pino and his team <br>
are **the** go to professionals. <br>   
For Business Brokers, Restaurants are perhaps the most vibrant sector. Thus, knowing were Restuarants are <br>
located and if there are areas of Restaurant concentration is very important information when we are searching <br>
for new clients.<br>         
Although Pino is a seasoned Professional and I am pretty sure that he knows the Vancouver landscape by heart, <br>
the information of Restaurants' locations will definitely be very important in the training and onboarding of <br>
new brokers.<br>   
Pino my friend this is for you!!!<br>

<font size = "6">**THE DATA**</font>

Foursquare is the most trusted, independent location data platform for understanding how people move through the real world.<br>   
From Foursquare we will get all the Restuarants in Vancouver and their coordinates, so that we can show their location on
a map of the area. <br> Showing every single Restaurant on the map will defeat the purpose. Instead we will group them by <br>
neighborhoods and show which neighborhoods have the biggest concentration. We use postcodes to define neighborhoouds. <br>   

The data for Postcodes,neighborhood names and coordinates were downloaded from the GeoNames Postal Code Dataset.<br>   
We combined the information to obtain a final list of Restaurants in each Vancouver Postal Code, and then grouped the neighborhoods
into 4 groups,<br>according to the Number of Restaurants.<br>   

We did not cross reference the Number of Restaurants with other data, e.g. frequency of visits or income of neighborhood inhabitants.<br>
There is no evidence to suggest that either of these parameters would influence if a Restaurant would potentially be for Sale.<br>   


<font size = "6">**THE CODE**</font><br>   
Pino you can skip this section!

*Import all the things that we need*

In [1]:
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

Solving environment: - 
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - defaults/noarch::ibm-wsrt-py37main-keep==0.0.0=2155
  - defaults/noarch::ibm-wsrt-py37main-main==custom=2155
  - conda-forge/linux-64::pytorch==1.8.0=cpu_py37ha70c682_1
done

# All requested packages already installed.

Libraries imported.


*Upload the file containing Canada's Postal Codes Information*

In [2]:
# The code was removed by Watson Studio for sharing.

In [3]:
#Get the neighbourhood coordiantes#
df_coordinates_raw = pd.read_csv(streaming_body_1, sep="\t", header=None)
df_coordinates_raw.head()

0    1                                   2        3   4    5   6    7   8   \
0  CA  T0A          Eastern Alberta (St. Paul)  Alberta  AB  NaN NaN  NaN NaN   
1  CA  T0B         Wainwright Region (Tofield)  Alberta  AB  NaN NaN  NaN NaN   
2  CA  T0C          Central Alberta (Stettler)  Alberta  AB  NaN NaN  NaN NaN   
3  CA  T0E            Western Alberta (Jasper)  Alberta  AB  NaN NaN  NaN NaN   
4  CA  T0G  North Central Alberta (Slave Lake)  Alberta  AB  NaN NaN  NaN NaN   

        9         10   11  
0  54.7660 -111.7174  6.0  
1  53.0727 -111.5816  6.0  
2  52.1431 -111.6941  5.0  
3  53.6758 -115.0948  5.0  
4  55.6993 -114.4529  6.0

In [4]:
#Select only the columns that we need#
df_coordinates=df_coordinates_raw[[1,3,2,9,10]]
df_coordinates = df_coordinates.rename(columns={1: 'Postal Code', 3: 'Borough', 2: 'Neighbourhood', 9: 'Latitude', 10: 'Longtitude'})
df_coordinates.tail()

Postal Code       Borough                     Neighbourhood  Latitude  \
1651         S9V  Saskatchewan                      Lloydminster   53.2835   
1652         S9X  Saskatchewan                       Meadow Lake   54.1335   
1653         Y0A         Yukon  Southeastern Yukon (Watson Lake)   61.5793   
1654         Y0B         Yukon       Central Yukon (Dawson City)   64.6450   
1655         Y1A         Yukon                        Whitehorse   60.7161   

      Longtitude  
1651   -110.0016  
1652   -108.4347  
1653   -131.1481  
1654   -137.5360  
1655   -135.0537

In [5]:
#Select the neighborhoods in Vancouver, only#
vancouver_data = df_coordinates[df_coordinates['Neighbourhood'].str.contains('Vancouver')]
vancouver_data.head()

Postal Code           Borough  \
154         V5K  British Columbia   
155         V5L  British Columbia   
156         V5M  British Columbia   
157         V5N  British Columbia   
158         V5P  British Columbia   

                                         Neighbourhood  Latitude  Longtitude  
154                 Vancouver (North Hastings-Sunrise)   49.2807   -123.0397  
155              Vancouver (North Grandview-Woodlands)   49.2795   -123.0667  
156  Vancouver (South Hastings-Sunrise / North Renf...   49.2600   -123.0398  
157  Vancouver (South Grandview-Woodlands / NE Kens...   49.2551   -123.0667  
158    Vancouver (SE Kensington / Victoria-Fraserview)   49.2220   -123.0683

In [6]:
vancouver_data.shape

(44, 5)

In [21]:
print("There are " + str(len(vancouver_data)) + " neighborhoods in Vancouver")

There are 44 neighborhoods in Vancouver


*Insert the Cordiantes of Vancouver*

In [8]:
address = 'Vancouver, CA'

latitude = 49.282729
longitude = -123.120738
print('The geograpical coordinate of Vancouver are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Vancouver are 49.282729, -123.120738.


In [22]:
# create map of Vancouver showing the different neighborhoods#
map_vancouver = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, label in zip(vancouver_data['Latitude'], vancouver_data['Longtitude'], vancouver_data['Neighbourhood']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_vancouver)  
    
map_vancouver

*Foursquare Credentials, cell deleted*

In [23]:
#Find all the Venues in Toronto#
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [24]:
vancouver_venues = getNearbyVenues(names=vancouver_data['Neighbourhood'],
                                   latitudes=vancouver_data['Latitude'],
                                   longitudes=vancouver_data['Longtitude']
                                  )

Vancouver (North Hastings-Sunrise)
Vancouver (North Grandview-Woodlands)
Vancouver (South Hastings-Sunrise / North Renfrew-Collingwood)
Vancouver (South Grandview-Woodlands / NE Kensington)
Vancouver (SE Kensington / Victoria-Fraserview)
Vancouver (South Renfrew-Collingwood)
Vancouver (Killarney)
Vancouver (East Mount Pleasant)
Vancouver (West Kensington / NE Riley Park-Little Mountain)
Vancouver (SE Riley Park-Little Mountain / SW Kensington / NE Oakridge / North Sunset)
Vancouver (SE Oakridge / East Marpole / South Sunset)
Vancouver (West Mount Pleasant / West Riley Park-Little Mountain)
Vancouver (East Fairview / South Cambie)
Vancouver (Strathcona / Chinatown / Downtown Eastside)
Vancouver (NE Downtown / Harbour Centre / Gastown / Yaletown)
Vancouver (Waterfront / Coal Harbour / Canada Place)
Vancouver (South West End)
Vancouver (North West End / Stanley Park)
Vancouver (West Fairview / Granville Island / NE Shaughnessy)
Vancouver (NW Shaughnessy / East Kitsilano / Quilchena)
Vanco

In [13]:
vancouver_venues.shape

(798, 7)

*Foursquare lists 798 venues in Vancouver. Let;s have a look at a few of them*

In [25]:
vancouver_venues.head()

Neighborhood  Neighborhood Latitude  \
0  Vancouver (North Hastings-Sunrise)                49.2807   
1  Vancouver (North Hastings-Sunrise)                49.2807   
2  Vancouver (North Hastings-Sunrise)                49.2807   
3  Vancouver (North Hastings-Sunrise)                49.2807   
4  Vancouver (North Hastings-Sunrise)                49.2807   

   Neighborhood Longitude                  Venue  Venue Latitude  \
0               -123.0397    The Fair at the PNE       49.282971   
1               -123.0397  Wooden Roller Coaster       49.281744   
2               -123.0397        Livestock Barns       49.284037   
3               -123.0397               Playland       49.281924   
4               -123.0397          Italia Bakery       49.280940   

   Venue Longitude                Venue Category  
0      -123.042109                          Fair  
1      -123.035128  Theme Park Ride / Attraction  
2      -123.039278                          Farm  
3      -123.036258                    Theme Park  
4      -123.045885                        Bakery

In [26]:
#group venues by neighbourhood#
vancouver_grouped_i = vancouver_venues.groupby('Neighborhood').count()
vancouver_grouped_i

Neighborhood Latitude  \
Neighborhood                                                                
North Vancouver East Central                                            1   
North Vancouver North Central                                           3   
North Vancouver Northwest                                              19   
North Vancouver Northwest Central                                       3   
North Vancouver South Central                                           4   
North Vancouver Southwest                                              12   
North Vancouver Southwest Central                                       1   
Vancouver (Bentall Centre)                                              9   
Vancouver (Central Kitsilano)                                          28   
Vancouver (Chaldecutt / South University Endowm...                      2   
Vancouver (Dunbar-Southlands / Musqueam)                                3   
Vancouver (East Fairview / South Cambie)                               24   
Vancouver (East Mount Pleasant)                                        23   
Vancouver (Killarney)                                                  18   
Vancouver (NE Downtown / Harbour Centre / Gasto...                     37   
Vancouver (NW Arbutus Ridge)                                            4   
Vancouver (NW Shaughnessy / East Kitsilano / Qu...                     22   
Vancouver (North Grandview-Woodlands)                                  28   
Vancouver (North Hastings-Sunrise)                                     24   
Vancouver (North West End / Stanley Park)                               8   
Vancouver (Pacific Centre)                                            100   
Vancouver (SE Kensington / Victoria-Fraserview)                        21   
Vancouver (SE Kerrisdale / SW Oakridge / West M...                     20   
Vancouver (SE Oakridge / East Marpole / South S...                      4   
Vancouver (SE Riley Park-Little Mountain / SW K...                     20   
Vancouver (SW Downtown)                                                55   
Vancouver (South Grandview-Woodlands / NE Kensi...                     17   
Vancouver (South Hastings-Sunrise / North Renfr...                     24   
Vancouver (South Renfrew-Collingwood)                                   6   
Vancouver (South Shaughnessy / NW Oakridge / NE...                      7   
Vancouver (South West End)                                            100   
Vancouver (Strathcona / Chinatown / Downtown Ea...                     15   
Vancouver (Waterfront / Coal Harbour / Canada P...                     97   
Vancouver (West Fairview / Granville Island / N...                      7   
Vancouver (West Kensington / NE Riley Park-Litt...                     22   
Vancouver (West Kitsilano / Jericho)                                    1   
Vancouver (West Mount Pleasant / West Riley Par...                      4   
West Vancouver South                                                    1   
West Vancouver Southeast                                                2   
West Vancouver West                                                     2   

                                                    Neighborhood Longitude  \
Neighborhood                                                                 
North Vancouver East Central                                             1   
North Vancouver North Central                                            3   
North Vancouver Northwest                                               19   
North Vancouver Northwest Central                                        3   
North Vancouver South Central                                            4   
North Vancouver Southwest                                               12   
North Vancouver Southwest Central                                        1   
Vancouver (Bentall Centre)                                               9   
Vancouver (Central Kitsilano)        

In [16]:
vancouver_grouped_i.shape

(40, 6)

*From Foursqaure we found venues from 40 neighborhoods<br>
Our Postal Codes Files has 44 neighborhoods<br>
This means that there are no venues on Foursquare for 3 neighborhoods*

*Let's look at all the different venue categories*

In [17]:
vancouver_venues['Venue Category'].unique()

array(['Fair', 'Theme Park Ride / Attraction', 'Farm', 'Theme Park',
       'Bakery', 'Beer Garden', 'Soccer Field', 'Park', 'Amphitheater',
       'Gas Station', 'Event Space', 'Sandwich Place',
       'Vietnamese Restaurant', 'Sushi Restaurant', 'Farmers Market',
       'Stadium', 'Bus Line', 'Inn', 'Pizza Place',
       'Portuguese Restaurant', 'Deli / Bodega', 'Coffee Shop',
       'Italian Restaurant', 'Theater', 'Fried Chicken Joint', 'Brewery',
       'Thai Restaurant', 'Café', 'Breakfast Spot', 'Japanese Restaurant',
       'Athletics & Sports', 'Thrift / Vintage Store',
       'Chinese Restaurant', 'Indian Restaurant', 'Asian Restaurant',
       'Electronics Store', 'Fish & Chips Shop', 'Convenience Store',
       'Pet Store', 'Liquor Store', 'Office', 'Grocery Store',
       'Paper / Office Supplies Store', 'Mexican Restaurant',
       'Insurance Office', 'Furniture / Home Store', 'Mobile Phone Shop',
       'Bus Stop', 'Metro Station', 'Storage Facility', 'Restaurant',
     

*Clean up the neighborhood name to remove the leading "Vancouver" and the parentheses*

In [28]:
vancouver_venues['Col1'] = vancouver_venues['Neighborhood'].str.strip('Vancouver')
vancouver_venues['Col3'] = vancouver_venues['Col1'].str.replace('(','')
vancouver_venues['Col4'] = vancouver_venues['Col3'].str.replace(')','')
vancouver_venues.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  from ipykernel import kernelapp as app
/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:3: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  app.launch_new_instance()


Neighborhood  Neighborhood Latitude  \
0  Vancouver (North Hastings-Sunrise)                49.2807   
1  Vancouver (North Hastings-Sunrise)                49.2807   
2  Vancouver (North Hastings-Sunrise)                49.2807   
3  Vancouver (North Hastings-Sunrise)                49.2807   
4  Vancouver (North Hastings-Sunrise)                49.2807   

   Neighborhood Longitude                  Venue  Venue Latitude  \
0               -123.0397    The Fair at the PNE       49.282971   
1               -123.0397  Wooden Roller Coaster       49.281744   
2               -123.0397        Livestock Barns       49.284037   
3               -123.0397               Playland       49.281924   
4               -123.0397          Italia Bakery       49.280940   

   Venue Longitude                Venue Category                       Col1  \
0      -123.042109                          Fair   (North Hastings-Sunrise)   
1      -123.035128  Theme Park Ride / Attraction   (North Hastings-Sunrise)   
2      -123.039278                          Farm   (North Hastings-Sunrise)   
3      -123.036258                    Theme Park   (North Hastings-Sunrise)   
4      -123.045885                        Bakery   (North Hastings-Sunrise)   

                       Col3                     Col4  
0   North Hastings-Sunrise)   North Hastings-Sunrise  
1   North Hastings-Sunrise)   North Hastings-Sunrise  
2   North Hastings-Sunrise)   North Hastings-Sunrise  
3   North Hastings-Sunrise)   North Hastings-Sunrise  
4   North Hastings-Sunrise)   North Hastings-Sunrise

In [30]:
vancouver_venues_2 = vancouver_venues.drop(['Col1','Col3'], axis=1)
vancouver_venues_2.head()

Neighborhood  Neighborhood Latitude  \
0  Vancouver (North Hastings-Sunrise)                49.2807   
1  Vancouver (North Hastings-Sunrise)                49.2807   
2  Vancouver (North Hastings-Sunrise)                49.2807   
3  Vancouver (North Hastings-Sunrise)                49.2807   
4  Vancouver (North Hastings-Sunrise)                49.2807   

   Neighborhood Longitude                  Venue  Venue Latitude  \
0               -123.0397    The Fair at the PNE       49.282971   
1               -123.0397  Wooden Roller Coaster       49.281744   
2               -123.0397        Livestock Barns       49.284037   
3               -123.0397               Playland       49.281924   
4               -123.0397          Italia Bakery       49.280940   

   Venue Longitude                Venue Category                     Col4  
0      -123.042109                          Fair   North Hastings-Sunrise  
1      -123.035128  Theme Park Ride / Attraction   North Hastings-Sunrise  
2      -123.039278                          Farm   North Hastings-Sunrise  
3      -123.036258                    Theme Park   North Hastings-Sunrise  
4      -123.045885                        Bakery   North Hastings-Sunrise

*Merge "vancouver_venues_2" with "vancouver_data" to get the Postal Codes*

In [31]:
final_merged = pd.merge(vancouver_venues_2, vancouver_data, left_on='Neighborhood', right_on='Neighbourhood', how='left')
final_merged.head()

Neighborhood  Neighborhood Latitude  \
0  Vancouver (North Hastings-Sunrise)                49.2807   
1  Vancouver (North Hastings-Sunrise)                49.2807   
2  Vancouver (North Hastings-Sunrise)                49.2807   
3  Vancouver (North Hastings-Sunrise)                49.2807   
4  Vancouver (North Hastings-Sunrise)                49.2807   

   Neighborhood Longitude                  Venue  Venue Latitude  \
0               -123.0397    The Fair at the PNE       49.282971   
1               -123.0397  Wooden Roller Coaster       49.281744   
2               -123.0397        Livestock Barns       49.284037   
3               -123.0397               Playland       49.281924   
4               -123.0397          Italia Bakery       49.280940   

   Venue Longitude                Venue Category                     Col4  \
0      -123.042109                          Fair   North Hastings-Sunrise   
1      -123.035128  Theme Park Ride / Attraction   North Hastings-Sunrise   
2      -123.039278                          Farm   North Hastings-Sunrise   
3      -123.036258                    Theme Park   North Hastings-Sunrise   
4      -123.045885                        Bakery   North Hastings-Sunrise   

  Postal Code           Borough                       Neighbourhood  Latitude  \
0         V5K  British Columbia  Vancouver (North Hastings-Sunrise)   49.2807   
1         V5K  British Columbia  Vancouver (North Hastings-Sunrise)   49.2807   
2         V5K  British Columbia  Vancouver (North Hastings-Sunrise)   49.2807   
3         V5K  British Columbia  Vancouver (North Hastings-Sunrise)   49.2807   
4         V5K  British Columbia  Vancouver (North Hastings-Sunrise)   49.2807   

   Longtitude  
0   -123.0397  
1   -123.0397  
2   -123.0397  
3   -123.0397  
4   -123.0397

*Drop unecessary columns and rename necessary onew to get a tidy dataframe*

In [32]:
final_merged_2 = final_merged.drop(['Neighbourhood', 'Borough', 'Latitude', 'Longtitude'], axis=1)
final_merged_2.head()

Neighborhood  Neighborhood Latitude  \
0  Vancouver (North Hastings-Sunrise)                49.2807   
1  Vancouver (North Hastings-Sunrise)                49.2807   
2  Vancouver (North Hastings-Sunrise)                49.2807   
3  Vancouver (North Hastings-Sunrise)                49.2807   
4  Vancouver (North Hastings-Sunrise)                49.2807   

   Neighborhood Longitude                  Venue  Venue Latitude  \
0               -123.0397    The Fair at the PNE       49.282971   
1               -123.0397  Wooden Roller Coaster       49.281744   
2               -123.0397        Livestock Barns       49.284037   
3               -123.0397               Playland       49.281924   
4               -123.0397          Italia Bakery       49.280940   

   Venue Longitude                Venue Category                     Col4  \
0      -123.042109                          Fair   North Hastings-Sunrise   
1      -123.035128  Theme Park Ride / Attraction   North Hastings-Sunrise   
2      -123.039278                          Farm   North Hastings-Sunrise   
3      -123.036258                    Theme Park   North Hastings-Sunrise   
4      -123.045885                        Bakery   North Hastings-Sunrise   

  Postal Code  
0         V5K  
1         V5K  
2         V5K  
3         V5K  
4         V5K

In [33]:
final_merged_3 = final_merged_2.drop(['Neighborhood'], axis=1)
final_merged_3= final_merged_3.rename(columns = {'Col4':'Neighborhood'})
final_merged_3.head()

Neighborhood Latitude  Neighborhood Longitude                  Venue  \
0                49.2807               -123.0397    The Fair at the PNE   
1                49.2807               -123.0397  Wooden Roller Coaster   
2                49.2807               -123.0397        Livestock Barns   
3                49.2807               -123.0397               Playland   
4                49.2807               -123.0397          Italia Bakery   

   Venue Latitude  Venue Longitude                Venue Category  \
0       49.282971      -123.042109                          Fair   
1       49.281744      -123.035128  Theme Park Ride / Attraction   
2       49.284037      -123.039278                          Farm   
3       49.281924      -123.036258                    Theme Park   
4       49.280940      -123.045885                        Bakery   

              Neighborhood Postal Code  
0   North Hastings-Sunrise         V5K  
1   North Hastings-Sunrise         V5K  
2   North Hastings-Sunrise         V5K  
3   North Hastings-Sunrise         V5K  
4   North Hastings-Sunrise         V5K

In [34]:
final_reordered = final_merged_3[['Postal Code','Neighborhood','Neighborhood Latitude','Neighborhood Longitude','Venue', 'Venue Latitude', 'Venue Longitude', 'Venue Category']]
final_reordered.head()

Postal Code             Neighborhood  Neighborhood Latitude  \
0         V5K   North Hastings-Sunrise                49.2807   
1         V5K   North Hastings-Sunrise                49.2807   
2         V5K   North Hastings-Sunrise                49.2807   
3         V5K   North Hastings-Sunrise                49.2807   
4         V5K   North Hastings-Sunrise                49.2807   

   Neighborhood Longitude                  Venue  Venue Latitude  \
0               -123.0397    The Fair at the PNE       49.282971   
1               -123.0397  Wooden Roller Coaster       49.281744   
2               -123.0397        Livestock Barns       49.284037   
3               -123.0397               Playland       49.281924   
4               -123.0397          Italia Bakery       49.280940   

   Venue Longitude                Venue Category  
0      -123.042109                          Fair  
1      -123.035128  Theme Park Ride / Attraction  
2      -123.039278                          Farm  
3      -123.036258                    Theme Park  
4      -123.045885                        Bakery

*Selecting only the data that have to do with Restaurants<br>   
Using only the "Venue Category" names that contain the word "Restaurant"<br>
leaves out Burger (joint), Fried Chicken (joint), Pizza (place) and diner<br>
which are really popular in business sales. So, let us add them as well.*   

In [39]:
vancouver_venues_new=final_reordered[final_reordered['Venue Category'].str.contains('Restaurant|Pizza|Burger|Fried Chicken|Diner')]
vancouver_venues_new.head()

Postal Code                Neighborhood  Neighborhood Latitude  \
12         V5K      North Hastings-Sunrise                49.2807   
13         V5K      North Hastings-Sunrise                49.2807   
22         V5K      North Hastings-Sunrise                49.2807   
23         V5K      North Hastings-Sunrise                49.2807   
26         V5L   North Grandview-Woodlands                49.2795   

    Neighborhood Longitude           Venue  Venue Latitude  Venue Longitude  \
12               -123.0397         Pho Don       49.280947      -123.045543   
13               -123.0397  Blowfish Sushi       49.281011      -123.045696   
22               -123.0397    Dijo's Pizza       49.277315      -123.038115   
23               -123.0397  Senhor Rooster       49.277320      -123.038085   
26               -123.0667        Pepino’s       49.279134      -123.070567   

           Venue Category  
12  Vietnamese Restaurant  
13       Sushi Restaurant  
22            Pizza Place  
23  Portuguese Restaurant  
26     Italian Restaurant

In [41]:
#Group the data by Neighborhood#
vancouver_venues_new['No of Restaurants']=vancouver_venues_new.groupby(["Neighborhood"])['Venue'].transform('count')
vancouver_venues_new.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  from ipykernel import kernelapp as app


Postal Code                Neighborhood  Neighborhood Latitude  \
12         V5K      North Hastings-Sunrise                49.2807   
13         V5K      North Hastings-Sunrise                49.2807   
22         V5K      North Hastings-Sunrise                49.2807   
23         V5K      North Hastings-Sunrise                49.2807   
26         V5L   North Grandview-Woodlands                49.2795   

    Neighborhood Longitude           Venue  Venue Latitude  Venue Longitude  \
12               -123.0397         Pho Don       49.280947      -123.045543   
13               -123.0397  Blowfish Sushi       49.281011      -123.045696   
22               -123.0397    Dijo's Pizza       49.277315      -123.038115   
23               -123.0397  Senhor Rooster       49.277320      -123.038085   
26               -123.0667        Pepino’s       49.279134      -123.070567   

           Venue Category  No of Restaurants  
12  Vietnamese Restaurant                  4  
13       Sushi Restaurant                  4  
22            Pizza Place                  4  
23  Portuguese Restaurant                  4  
26     Italian Restaurant                 11

In [43]:
#Create dataframe with single row per neighbourhood#
new_restaurants = vancouver_venues_new.drop_duplicates('Neighborhood')
new_restaurants.head()

Postal Code                                       Neighborhood  \
12          V5K                             North Hastings-Sunrise   
26          V5L                          North Grandview-Woodlands   
59          V5M   South Hastings-Sunrise / North Renfrew-Collin...   
95          V5P                SE Kensington / Victoria-Fraserview   
114         V5R                          South Renfrew-Collingwood   

     Neighborhood Latitude  Neighborhood Longitude  \
12                 49.2807               -123.0397   
26                 49.2795               -123.0667   
59                 49.2600               -123.0398   
95                 49.2220               -123.0683   
114                49.2397               -123.0407   

                             Venue  Venue Latitude  Venue Longitude  \
12                         Pho Don       49.280947      -123.045543   
26                        Pepino’s       49.279134      -123.070567   
59   Vallarta's Mexican Restaurant       49.262232      -123.044736   
95           Tibisti Foods & Grill       49.220397      -123.065821   
114             Chili Pepper House       49.236270      -123.041805   

                Venue Category  No of Restaurants  
12       Vietnamese Restaurant                  4  
26          Italian Restaurant                 11  
59          Mexican Restaurant                  6  
95   Middle Eastern Restaurant                  9  
114           Asian Restaurant                  1

In [45]:
#Keep only the columns that we need and print the entire dataframe#
final_data = new_restaurants[['Postal Code', 'Neighborhood','Neighborhood Longitude', 'Neighborhood Latitude', 'No of Restaurants']]
final_data

Postal Code                                       Neighborhood  \
12          V5K                             North Hastings-Sunrise   
26          V5L                          North Grandview-Woodlands   
59          V5M   South Hastings-Sunrise / North Renfrew-Collin...   
95          V5P                SE Kensington / Victoria-Fraserview   
114         V5R                          South Renfrew-Collingwood   
120         V5S                                          Killarney   
139         V5T                                East Mount Pleasant   
164         V5V    West Kensington / NE Riley Park-Little Mountain   
184         V5W   SE Riley Park-Little Mountain / SW Kensington...   
203         V5X          SE Oakridge / East Marpole / South Sunset   
207         V5Y   West Mount Pleasant / West Riley Park-Little ...   
212         V5Z                       East Fairview / South Cambie   
240         V6A         Strathcona / Chinatown / Downtown Eastside   
253         V6B   NE Downtown / Harbour Centre / Gastown / Yale...   
296         V6C           Waterfront / Coal Harbour / Canada Place   
386         V6E                                     South West End   
493         V6H   West Fairview / Granville Island / NE Shaughn...   
501         V6J        NW Shaughnessy / East Kitsilano / Quilchena   
523         V6K                                  Central Kitsilano   
549         V6L                                   NW Arbutus Ridge   
553         V6M   South Shaughnessy / NW Oakridge / NE Kerrisda...   
560         V6N                       Dunbar-Southlands / Musqueam   
565         V6P         SE Kerrisdale / SW Oakridge / West Marpole   
589         V6Z                                        SW Downtown   
647         V7L                      North Vancouver South Central   
653         V7P                          North Vancouver Southwest   
671         V7R                          North Vancouver Northwest   
688         V7W                                West Vancouver West   
693         V7X                                     Bentall Centre   
716         V7Y                                     Pacific Centre   

     Neighborhood Longitude  Neighborhood Latitude  No of Restaurants  
12                -123.0397                49.2807                  4  
26                -123.0667                49.2795                 11  
59                -123.0398                49.2600                  6  
95                -123.0683                49.2220                  9  
114               -123.0407                49.2397                  1  
120               -123.0380                49.2175                  4  
139               -123.0923                49.2620                  8  
164               -123.0913                49.2480                  8  
184               -123.0917                49.2327                 12  
203               -123.0979                49.2156                  2  
207               -123.1104                49.2492                  2  
212               -123.1210                49.2475                  5  
240               -123.0908                49.2779                  4  
253               -123.1139                49.2788                 11  
296               -123.1158                49.2866                 18  
386               -123.1298                49.2833                 45  
493               -123.1322                49.2559                  2  
501               -123.1460                49.2603                  7  
523               -123.1648                49.2646                  9  
549               -123.1660                49.2497                  2  
553               -123.1451                49.2344                  4  
560               -123.1890                49.2302                  2  
565               -123.1396                49.2151                 11  
589               -123.1300                49.2767                 24  
647               -123.0576      

*And the top 5 areas with the most Restaurants are:*

In [46]:
final_data_sorted = final_data.sort_values(by=['No of Restaurants'], ascending=False)
final_data_sorted.head()

Postal Code                                       Neighborhood  \
386         V6E                                     South West End   
716         V7Y                                     Pacific Centre   
589         V6Z                                        SW Downtown   
296         V6C           Waterfront / Coal Harbour / Canada Place   
184         V5W   SE Riley Park-Little Mountain / SW Kensington...   

     Neighborhood Longitude  Neighborhood Latitude  No of Restaurants  
386               -123.1298                49.2833                 45  
716               -123.1190                49.2819                 30  
589               -123.1300                49.2767                 24  
296               -123.1158                49.2866                 18  
184               -123.0917                49.2327                 12

*Split the Neighborhoods into 4 quartiles, based on the Number of Restaurants*

In [47]:
final_data['No of Restaurants Quartiles'] = pd.qcut(final_data['No of Restaurants'], 4, labels = False)
final_data.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if __name__ == '__main__':


Postal Code                                       Neighborhood  \
12          V5K                             North Hastings-Sunrise   
26          V5L                          North Grandview-Woodlands   
59          V5M   South Hastings-Sunrise / North Renfrew-Collin...   
95          V5P                SE Kensington / Victoria-Fraserview   
114         V5R                          South Renfrew-Collingwood   

     Neighborhood Longitude  Neighborhood Latitude  No of Restaurants  \
12                -123.0397                49.2807                  4   
26                -123.0667                49.2795                 11   
59                -123.0398                49.2600                  6   
95                -123.0683                49.2220                  9   
114               -123.0407                49.2397                  1   

     No of Restaurants Quartiles  
12                             1  
26                             3  
59                             2  
95                             2  
114                            0

***Create a map of Vancouver***

In [48]:
map_vancouver_2 = folium.Map(location=[latitude, longitude], zoom_start=11)
map_vancouver_2

***Colour code the 4 Groups and add the neighborhoods on the map
a lager label indicates a bigger number of restaurants***

In [49]:
colordict = {0: 'blue', 1: 'darkgreen', 2: 'orange', 3: 'red'}

In [53]:
for lat, lon, Quartiles, Number, Location in zip(final_data['Neighborhood Latitude'], final_data['Neighborhood Longitude'], final_data['No of Restaurants Quartiles'], final_data['No of Restaurants'], final_data['Neighborhood']):
        folium.CircleMarker(
        [lat, lon],
        radius=5+3*Quartiles,
        popup = (str(Location) + '<br>'
                 'Number of Restaurants: ' + str(Number)),
        color='b',
        key_on = Quartiles,
        threshold_scale=[0,1,2,3],
        fill_color=colordict[Quartiles],
        fill=True,
        fill_opacity=0.7
        ).add_to(map_vancouver_2)
map_vancouver_2

***Pino my friend all the action is concentrated in the South West/Harbor Area!<br>
(Almost) Nothing to do North of the Harbour***